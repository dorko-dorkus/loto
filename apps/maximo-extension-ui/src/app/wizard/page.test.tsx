import { cleanup, fireEvent, render, screen } from '@testing-library/react';
import { afterEach, beforeEach, expect, test } from 'vitest';
import Page from './page';

beforeEach(() => {
  window.localStorage.clear();
});

afterEach(() => {
  cleanup();
});

test('matches snapshot', () => {
  const { container } = render(<Page />);
  expect(container.firstChild).toMatchSnapshot();
});

test('initializes from and persists to localStorage', async () => {
  window.localStorage.setItem('wizard-step', '1');
  window.localStorage.setItem(
    'wizard-data',
    JSON.stringify({ name: 'Bob', age: '42' })
  );
  render(<Page />);
  const age = screen.getByLabelText('age');
  expect(age).toHaveValue('42');

  fireEvent.change(age, { target: { value: '30' } });
  fireEvent.click(screen.getByText('Next'));

  expect(window.localStorage.getItem('wizard-step')).toBe('2');
  const stored = window.localStorage.getItem('wizard-data');
  expect(stored && JSON.parse(stored)).toEqual({ name: 'Bob', age: '30' });
});

