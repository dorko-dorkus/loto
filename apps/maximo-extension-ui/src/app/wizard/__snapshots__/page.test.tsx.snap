// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches snapshot 1`] = `
<main>
  <ol
    class="flex gap-2"
  >
    <li
      class="rounded px-3 py-1 bg-[var(--mxc-topbar-bg)] text-[var(--mxc-topbar-fg)]"
    >
      Step One
    </li>
    <li
      class="rounded px-3 py-1 bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]"
    >
      Step Two
    </li>
    <li
      class="rounded px-3 py-1 bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]"
    >
      Step Three
    </li>
    <li
      class="rounded px-3 py-1 bg-[var(--mxc-sheet)] text-[var(--mxc-fg-subtle)]"
    >
      Export
    </li>
  </ol>
  <div>
    <label>
      Name
      <input
        aria-label="name"
        value=""
      />
    </label>
    <button>
      Next
    </button>
  </div>
</main>
`;
